csfieldguide/chapters/content/en/coding-compression/sections/general-purpose.md

```diff
@@ -43,27 +23,308 @@ Here are some of the bits from the example earlier in this chapter; you can past
+| 3    | 2    | M      | 5                   | AM                | M?                          |
+| 4    | 3    | S      | 6                   | MS                | S?                          |
+| 5    | 5    | AM     | 7                   | SA                | AM?                         |
+
+In step 5 we have to decode 5. Recall that 5 wasn’t in the initial dictionary, but since we added it in step 2 are able to decode it.
+
+| Step | Code | Output | New Dict Entry Code | New Dict Sequence | Next Proposed Dict Sequence |
+|------|------|--------|---------------------|-------------------|-----------------------------|
+| 6    | 7    | SA     | 8                   | AMS               | SA?                         |
+| 7    | 2    | M      | 9                   | SAM               | M?                          |
+| 8    | 4    | IA     | 10                  | MI                | IA?                         |
+| 9    | 2    | M      | 11                  | IAM               | M?                          |
+
+Once we have decoded the whole message we now have the following string:
+
+```
+IAMSAMSAMIAM
+```
+
+So we successfully encoded and decoded the message!
+And this is the dictionary that the decoder created:
+
+| Code | Bits         | Sequence |
+|------|--------------|----------|
+| 0    | 000000000000 | I        |
+| 1    | 000000000001 | A        |
+| 2    | 000000000010 | M        |
+| 3    | 000000000011 | S        |
+| 4    | 000000000100 | IA       |
+| 5    | 000000000101 | AM       |
+| 6    | 000000000110 | MS       |
+| 7    | 000000000111 | SA       |
+| 8    | 000000001000 | AMS      |
+| 9    | 000000001001 | SAM      |
+| 10   | 000000001010 | MI       |
+| 11   | 000000001011 | IAM      |
+
+Notice that it is exactly the same dictionary as the encoder created - the dictionary has been reconstructed without having to send it to the decoder!
+
+
+{panel type="extra-for-experts" subtitle="true"}
+
+# Extra for Experts
+
+## Other languages
+
+But what about different languages? Since the dictionary builds up based on the characters it has seen, it will simply adapt to whatever language is being used, whether it is English, French, Chinese, or even a computer programming language. Each language will have its own common substrings, and these will end up in the dictionary.
 
 {panel end}
 
-{panel type="curiosity"}
+{panel type="extra-for-experts" subtitle="true"}
+
+# Extra for Experts
 
-# ZL or LZ compression?
+## Representing the codes in fewer than 12 bits
 
-The method we have described here is named "Ziv-Lempel" compression after Jacob Ziv and Abraham Lempel, the two computer scientists who invented it in the 1970s.
-Unfortunately someone mixed up the order of their names when they wrote an article about it, and called it "LZ" compression instead of "ZL" compression.
-So many people copied the mistake that Ziv and Lempel’s method is now usually called "LZ compression"!
```

Comment:
This still seems like a good curiosity, does it need to be removed?

Reply:
doesn't really fit in anymore, especially since LZ/ZL isn't the main focus of the section anymore.